src/TestFramework/Coverage/ProxyTrace.php

```diff
@@ -79,15 +79,12 @@ public function __construct(SplFileInfo $sourceFile, iterable $lazyTestLocations
         $this->lazyTestLocations = $lazyTestLocations;
     }
 
-    public function getSplFileInfo(): SplFileInfo
+    public function getSourceFileInfo(): SplFileInfo
     {
         return $this->sourceFile;
     }
 
-    /**
-     * Used by RealPathFilterIterator
-     */
-    public function getRealPath(): string
+    public function getSourceRealPath(): string
```

Comment:
`RealPathFilterIterator` wants exactly `getRealPath()` because it expects `SplFileInfo`. Here we're duck-typing `ProxyTrace` to match.

Reply:
I see, welp definitely need to add a test for that then

Reply:
Alternatively we can update `RealPathFilterIterator`'s `current()` method to handle these differences for us. It'll require yet another SplFileInfo-like wrapper, and tbh I can't say it looks exactly better the duck-typing approach we take here.

Reply:
The duck-typing is fine it just needed a test which was not hard to do based on your code :)